pkgs/development/python-modules/nfstream/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib
+, fetchPypi
+, buildPythonPackage
+, buildPythonApplication
+, psutil
+, numpy
+, dpkt
+, cffi
+, pandas
+}:
+
+buildPythonPackage rec {
+  pname = "nfstream";
+  version = "6.3.5";
+  format = "wheel";
+
+  src = fetchPypi {
+    inherit pname version;
+    format = "wheel";
```

Comment:
Can't we fetch source?

Reply:
I tried a while ago, without success..
https://discourse.nixos.org/t/python-package-wheel-no-matching-distribution-found/15659/31

Reply:
you want something like:
```nix
  postPatch = ''
    substituteInPlace files.. \
      --replace "/usr/local/include/ndpi" "${ndpi}/include/ndpi"
  '';
```

Reply:
Sorry for the delay :D
I tried to build it from source but got a bunch of errors regarding one of the C files.
`with import <nixpkgs> {};
with python39Packages;

buildPythonPackage rec {
  pname = "nfstream";
  version = "6.4.2";

  src = fetchFromGitHub {
    owner = "${pname}";
    repo = "${pname}";
    rev = "v${version}";
    sha256 = "sha256-5aYgGBUBbAq1n29r633RYNU7f6aTVItgEodk8sYmpH4=";
  };

  nativeBuildInputs = [
    autoconf automake libtool pkg-config flex bison gettext libjson
  ];

  strictDeps = false;
  
  buildInputs = [
    pkgsStatic.libgpgerror
    pkgsStatic.libgcrypt
    libpcap
    ndpi
    pytest pytest-runner
  ];

  postPatch = ''
    substituteInPlace setup.py \
      --replace "/usr/local/include/ndpi" "${ndpi}/include/ndpi" \
      --replace "/usr/local/lib/libpcap.a" "${libpcap}/lib/libpcap.a" \
      --replace "/usr/local/lib/libndpi.a" "${ndpi}/lib/libndpi.a" \
      --replace "/usr/local/lib/libgcrypt.a" "${pkgsStatic.libgcrypt}/lib/libgcrypt.a" \
      --replace "/usr/local/lib/libgpg-error.a" "${pkgsStatic.libgpg-error}/lib/libgpg-error.a"
  '';
      
  propagatedBuildInputs = with python39Packages; [ psutil numpy dpkt cffi pandas  ];

  meta = with lib; {
    description = "Python Framework for network data analytics";
    homepage = "https://github.com/nfstream/nfstream";
    license = licenses.lgpl3;
    maintainers = with maintainers; [ heph2 ];
  };
}
`

And here's the errors:
https://clbin.com/hB8hO

Reply:
Thats really one error. We have ndpi 4.2 and for that to work you need some commits from https://github.com/nfstream/nfstream/commits/master. I would suggest to try fetching the latest commit from master and see if that works.

Also in pythonPackages you are not allowed to input python*Packages.

Reply:
Also please drop pytest-runner and put pytest into checkInputs. strictDeps should also not be turned off. libjson is most likely also a buildInput.